packages/@storefront/sort/src/sort/index.ts

```diff
@@ -34,21 +34,35 @@ class Sort {
     this.updateSorts();
   }
 
+  onUpdate() {
+    this.state = {
+      ...this.state,
+      sorts: this.extractSorts(),
+    };
+  }
+
   updateSorts = () => this.set({ sorts: this.extractSorts() });
 
   extractSorts() {
     let sorts;
+    let labels;
     switch (this.props.storeSection) {
       case StoreSections.PAST_PURCHASES:
         sorts = this.select(Selectors.pastPurchaseSort);
+        labels = this.props.pastPurchasesLabels && this.props.pastPurchasesLabels.length
```

Comment:
`this.props.pastPurchaseLabels` and `this.props.labels` seem like good candidate for destructuring into local constants.

Reply:
Will investigate.

Reply:
This feedback item may have been addressed by the introduction of `extractLabels()`. Please re-review.